app/org/maproulette/controllers/api/TaskBundleController.scala

```diff
@@ -118,21 +118,42 @@ class TaskBundleController @Inject() (
     * @param reviewStatus The review status id to set the task's review status to
     * @param comment      An optional comment to add to the task
     * @param tags         Optional tags to add to the task
+    * @param newTaskStatus  Optional new taskStatus to change on all tasks in bundle
     * @return 400 BadRequest if task with supplied id not found.
     *         If successful then 200 NoContent
     */
   def setBundleTaskReviewStatus(
       id: Long,
       reviewStatus: Int,
       comment: String = "",
-      tags: String = ""
+      tags: String = "",
+      newTaskStatus: String = ""
   ): Action[AnyContent] = Action.async { implicit request =>
     this.sessionManager.authenticatedRequest { implicit user =>
-      val tasks = this.dalManager.taskBundle.getTaskBundle(user, id).tasks match {
+      var tasks = this.dalManager.taskBundle.getTaskBundle(user, id).tasks match {
         case Some(t) => t
         case None    => throw new InvalidException("No tasks found in this bundle.")
       }
 
+      // If the mapper wants to change the task status while revising the task after review
```

Comment:
You should just move all this code up into the section above by "case Some(t) => ..." 

Then you can keep it as val. If you ever find yourself using a var in Scala you should also look at your code and see if there is a better way to approach it. Scala's methodology moves more towards a functional approach which tries to steer clear of mutable objects.

Reply:
Refactored to remove var